### PR TITLE
Branch3

### PR DESCRIPTION
**Pull request by DadizhaO**

### CODE REVIEW

File: src/main/java/HM05/FruitMarket.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public List<Fruits> getFreshFruitsByKind(KindsOfFruits kind, LocalDate date) {
+        List<Fruits> result = listOfFruits.stream().
+                filter(a -> a.getKinds().equals(kind) && a.shelfLife(date)).collect(Collectors.toList());
+        result.forEach(System.out::println);
+        return result;
+    }
+
+    public List<Fruits> changePrice(KindsOfFruits kind, int percent) {
```
Comment (Taniann) [inline src/main/java/HM05/FruitMarket.java:50]: this method must work with few types of fruits. I mean you need to pass list of KindOfFruits or KindOfFruits... kind

File: src/main/java/HM05/FruitMarket.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    public List<Fruits> getFreshFruits(LocalDate date) {
+        List<Fruits> listOfFresh = new ArrayList<>();
+        for (Fruits fruit : listOfFruits) {
+            if (fruit.shelfLife(date)) {
+                listOfFresh.add(fruit);
+            }
+        }
+        listOfFresh.forEach(System.out::println);
```
Comment (Taniann) [inline src/main/java/HM05/FruitMarket.java:27]: better to call method println in external class, and print List that you return

File: src/main/java/HM05/FruitMarket.java (hunk 1 above)
Comment (Taniann) [reply src/main/java/HM05/FruitMarket.java:50]: this method implementation isn`t done exactly as task requires

**Review by Taniann — CHANGES_REQUESTED** [review]

File: src/main/java/HM04/Bank.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package HM04;
+
+
+class Bank <T extends BasicInterface> {
```
Comment (SerhiiPasko) [inline src/main/java/HM04/Bank.java:4]: Way you add Bank logic there?

File: src/main/java/HM05/FruitMarket.java (hunk 2 above)
Comment (SerhiiPasko) [reply src/main/java/HM05/FruitMarket.java:27]: Same to other

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments. Only one serious - you didnt do one of the tasks. It is easy restore, but you didnt create pull request in time

**Pull request merged**